jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/deadlines/notifications/impl/NotificationListenerManager.java

```diff
@@ -98,6 +101,21 @@ public void registerAdditionalNotificationListener(List<NotificationListener> ad
     public List<NotificationListener> getNotificationListeners() {
         return listeners;
     }
+
+    public void broadcast(TaskContext taskContext, NotificationEvent event, UserInfo userInfo) {
+        IdentityProvider identityProvider = (IdentityProvider) taskContext.get(EnvironmentName.IDENTITY_PROVIDER);
+        if(identityProvider != null) {
+            identityProvider.setContextIdentity(System.getProperty("org.jbpm.ht.admin.user", "Administrator"));
```

Comment:
The  problem here is those operations are not executed by anybody (deadlines and reminders so you need to set a context already(there is identity manager at runtime). 
Due to the nature of commands you don't know who is setting what so it is better to be safe than sorry here allowing a better check in the future. so you push the name and then you remove it from context identity at the end.
I set the name because it does make more sense to me to put the task administrator. We could tie to another variable But I don't see it necessary. We always can change that. So yes in this case we are always overriding the context from an anonymous user to Administrator which is exactly the problem. Anonymous does not have permission to modify user tasks.

Reply:
thanks for clarifying